lib/resty/openssl/revoked.lua

```diff
@@ -0,0 +1,85 @@
+local ffi = require "ffi"
+local bn_lib = require("resty.openssl.bn")
+require("resty.openssl.include.x509.crl")
+require("resty.openssl.include.x509.revoked")
+local NID_crl_reason = 141
+local C = ffi.C
+local ffi_gc = ffi.gc
+local _M = {}
+local revoked_ptr_ct = ffi.typeof('X509_REVOKED *')
+
+--- Creates new instance of X509_REVOKED data
+-- @tparam bn|number sn Serial number as number or bn instance
+-- @tparam number time Revocation time
+-- @tparam number reason Revocation reason
+-- @treturn table instance of the module or nil
+-- @treturn[opt] string Returns optional error message in case of error
+function _M.new(sn, time, reason)
+    --- only convert to bn if it is number
+    if type(sn) == "number"then
+        sn = bn_lib.new(sn)
+    end
+    local revoked = C.X509_REVOKED_new()
+    ffi_gc(revoked, C.X509_REVOKED_free)
+
+    time = C.ASN1_TIME_set(nil, time)
+    if time == nil then
+        return nil, "x509.revoked.new: ASN1_TIME_set() failed"
+    end
+    ffi_gc(time, C.ASN1_ENUMERATED_free)
```

Comment:
should we free it with `ASN1_STRING_free` instead?

Reply:
This line is not needed